src/bn256/fq6.rs

```diff
@@ -513,7 +508,7 @@ pub const FROBENIUS_COEFF_FQ6_C1: [Fq2; 6] = [
 ];
 
 pub const FROBENIUS_COEFF_FQ6_C2: [Fq2; 6] = [
-    // Fq2(u + 1)**(((2q^0) - 2) / 3)
+    // Fq2(u + 9)**(((2q^0) - 2) / 3)
```

Comment:
Can you elaborate where's this 9 coming from?

Reply:
The previous value is the cubic non_residue for `Bls12-381`. The values for the field extensions of Bn are :
```
   Fp2  = Fp[X]  / <X^2 + 1>             alpha^2 = -1
   Fp6  = Fp2[X] / <X^3 - (alpha + 9>    beta^3  = alpha + 9
   Fp12 = Fp6[X] / <X^2 - beta>          gamma^2 = beta
```